daml/ContingentClaims/Util.daml

```diff
@@ -27,11 +31,21 @@ enum' (i, Cofree _ (GiveF f)) = CofreeF i (GiveF (succ i, f))
 enum' (i, Cofree _ (AndF f@(Cofree depth _) f')) = CofreeF i (AndF (succ i, f) (succ (i + depth), f'))
 enum' (i, Cofree _ (OrF f@(Cofree depth _) f')) = CofreeF i (OrF (succ i, f) (succ (i + depth), f'))
 enum' (i, Cofree _ (CondF p f@(Cofree depth _) f')) = CofreeF i (CondF p (succ i, f) (succ (i + depth), f'))
+enum' (i, Cofree _ (AnytimeF p f)) = CofreeF i (AnytimeF p (succ i, f))
+enum' (i, Cofree _ (UntilF p f)) = CofreeF i (UntilF p (succ i, f))
 
 -- Enumerate each node in the tree, starting from from 'zero'
 enumerateFrom : Int -> Claim f t x a -> Cofree (ClaimF f t x a) Int
 enumerateFrom zero = ana enum' . (zero, ) . synthesize subTreeSize'
 
+fixings : Claim f t x a -> [t]
+fixings = cata fixings'
+
+--TODO should fail if dates will never be executed
+fixings' : ClaimF f t x a [t] -> [t]
+fixings' (WhenF (O.TimeGte t) ts) = t :: ts
+fixings' claim = fold claim
```

Comment:
If I understand this function correctly, it pulls out dates at which things happen. Shouldn't the match then be on time equality, ie. `O.TimeEqu`, instead of `O.TimeGte`?

Reply:
There's no time equality; as explained in my previous comment, the semantics of `<=` vary for `When`, which is a point operator, and `Anytime` which is a region operator.

I suppose we could add a `TimeEqu`, but it wouldn't behave any differently from `TimeGte`, and we'd need to write an extra case for any function we implement on `Claim`. What do you think @georg-da ? If you're confused by this, so will other people be too.